clippy_utils/src/lib.rs

```diff
@@ -1488,10 +1489,45 @@ pub fn match_function_call<'tcx>(
 /// Checks if `Ty` is normalizable. This function is useful
 /// to avoid crashes on `layout_of`.
 pub fn is_normalizable<'tcx>(cx: &LateContext<'tcx>, param_env: ty::ParamEnv<'tcx>, ty: Ty<'tcx>) -> bool {
-    cx.tcx.infer_ctxt().enter(|infcx| {
+    is_normalizable_helper(cx, param_env, ty, &mut HashMap::new())
```

Comment:
Wouldn't it be easier to use [`TyS::walk`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_middle/ty/struct.TyS.html#method.walk) here (with `any`)?

Reply:
Now that I know that exists, yes it would. I must have missed that before.